### PR TITLE
implement the function search compact design

### PR DESCRIPTION
**Pull request by clarkbw**
Associated Issue:  #2007

### Summary of Changes

* implement the compact design for function search results

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] Command-P works as normal
- [x] cmd+shift+o shows results in a new layout

### Screenshots/Videos (OPTIONAL)

screen shot 2017-02-17 at 2 24 59 pm

### CODE REVIEW

**Comment by clarkbw** [comment]
You'll notice that I went with the right hand side line numbers in a column on the right hand side but all aligned to the left so the `:` is vertically aligned.  This requires a bit more complex flex layout to achieve but I think its better and I believe we wanted to use it other places.

**Comment by jasonLaster** [comment]
This looks great and the cider is straight forward. 

One thought if you have time is that it would be nice to tweak the selected state. 

I'll test soon, but if you could take some screenshots of function search with the dark theme and source search that would help. With source search we just want to prevent regressions

**Comment by clarkbw** [comment]
Yes!  Good call☝️ 

I can back around to this Monday

**Comment by clarkbw** [comment]
The <kbd>cmd</kbd>+<kbd>P</kbd> search looks fine with these changes in both light and dark themes.

Light     | Dark
--------- | ---------
screen shot 2017-02-20 at 2 34 57 pm | screen shot 2017-02-20 at 2 34 36 pm

**Comment by clarkbw** [comment]
The <kbd>cmd</kbd>+<kbd>shift</kbd>+<kbd>o</kbd> search has some issues with the dark theme that I'll look into.

Light     | Dark
--------- | ---------
screen shot 2017-02-20 at 2 40 02 pm | screen shot 2017-02-20 at 2 40 38 pm

**Comment by clarkbw** [comment]
DOCKER!!! 🙊

**Comment by jasonLaster** [comment]
lets see if my branch works: https://circleci.com/gh/jasonLaster/debugger.html/2139

File: src/components/SourceSearch.js [hunk 1]
```diff
@@ -96,18 +96,21 @@ const Search = React.createClass({
   },
 
   render() {
-    return this.props.searchOn ?
-      dom.div({ className: "search-container" },
+    if (!this.props.searchOn) {
+      return null;
+    }
```
Comment (jasonLaster) [inline src/components/SourceSearch.js]: 👍

**Comment by jasonLaster** [comment]
Tests on my branch were green. We can always make follow up improvements

**Pull request merged**